check_diff/tests/git.rs

```diff
@@ -0,0 +1,25 @@
+use check_diff::clone_git_repo;
+
+use std::fs;
+use tempfile::Builder;
+
+#[test]
+fn clone_repo_test() {
+    // Creates an empty directory in the current working directory
+    let dir = Builder::new().tempdir_in("").unwrap();
+    let sample_repo = "https://github.com/rust-lang/rustfmt.git";
+    let dest_path = dir.path();
+    let result = clone_git_repo(sample_repo, dest_path);
+    assert_eq!(result.is_ok(), true);
+    let directory = fs::read_dir(dest_path);
+    // check that the directory is non-empty
+    assert_eq!(directory.iter().next().is_none(), false);
+    // check whether a .git folder exists in directory
+    let mut is_git_present = false;
+    for path in directory.unwrap() {
+        if path.unwrap().file_name() == ".git" {
+            is_git_present = true;
+        }
+    }
+    assert_eq!(is_git_present, true);
```

Comment:
Instead of looping we can test this more directly with [`PathBuf::exists`](https://doc.rust-lang.org/std/path/struct.Path.html#method.exists):

```suggestion
    // check whether a .git folder exists after cloning the repo
    let git_repo = dest_path.join(".git");
    assert!(git_repo.exists());
```